common/signal_processing/documentation/ButterworthFilter.md

```diff
@@ -0,0 +1,127 @@
+of the filter using;
+
+    bf.computeContinuousTimeTF();
+
+The computed continuous-time transfer function roots can be printed on the screen using the methods;
+
+    bf.PrintFilter_ContinuousTimeRoots();
+    bf.PrintContinuousTimeTF();
+
+The resulting screen output for a 5th order filter is demonstrated below.
+
+     Roots of Continous Time Filter Transfer Function Denominator are : 
+    -0.585518 + j 1.80204
+    -1.53291 + j 1.11372
+    -1.89478 + j 2.32043e-16
+    -1.53291 + j -1.11372
+    -0.585518 + j -1.80204
+    
+    
+    The Continuous-Time Transfer Function of the Filter is ;
+    
+                                       24.422 
+    -------------------------------------------------------------------------------
+    1.000 *s[5] + 6.132 *s[4] + 18.798 *s[3] + 35.619 *s[2] + 41.711 *s[1] + 24.422 
+
+#### Discrete Time Transfer Function (Difference Equations)
+
+The digital filter equivalent of the continuous-time definitions is produced by using the bilinear transformation.
+When creating the discrete-time function of the ButterworthFilter object, its Numerator (Bn) and Denominator (An
+) coefficients are stored in a vector of filter order size N.
+
+The discrete transfer function method is called using ;
+
+    bf.computeDiscreteTimeTF();
+    bf.PrintDiscreteTimeTF();
+
+The results are printed on the screen like;
+The Discrete-Time Transfer Function of the Filter is ;
+
+    0.191 *z[-5] + 0.956 *z[-4] + 1.913 *z[-3] + 1.913 *z[-2] + 0.956 *z[-1] + 0.191
+    --------------------------------------------------------------------------------
+    1.000 *z[-5] + 1.885 *z[-4] + 1.888 *z[-3] + 1.014 *z[-2] + 0.298 *z[-1] + 0.037
+
+and the associated difference coefficients An and Bn by withing a struct ;
+
+    sDifferenceAnBn AnBn = bf.getAnBn();
+
+The difference coefficients appear in the filtering equation in the form of.
+
+    An * Y_filtered = Bn * Y_unfiltered  
+
+To filter a signal given in a vector form ;
+
+#### Calling Filter by a specified cut-off and sampling frequencies [in Hz]
+
+The Butterworth filter can also be created by defining the desired order (N), a cut-off frequency (fc in [Hz]), and a
+sampling frequency (fs in [Hz]). In this method, the cut-off frequency is pre-warped with respect to the sampling
+frequency [1, 2] to match the continuous and digital filter frequencies.
+
+The filter is prepared by the following calling options;
+
+     // 3rd METHOD defining a sampling frequency together with the cut-off fc, fs
+     bf.setOrder(2);
+     bf.setCuttoffFrequency(10, 100);
+
+At this step, we define a boolean variable whether to use the pre-warping option or not.
+
+    // Compute Continuous Time TF
+    bool use_sampling_frequency = true;
+    bf.computeContinuousTimeTF(use_sampling_frequency);
+    bf.PrintFilter_ContinuousTimeRoots();
+    bf.PrintContinuousTimeTF();
+
+    // Compute Discrete Time TF
+    bf.computeDiscreteTimeTF(use_sampling_frequency);
+    bf.PrintDiscreteTimeTF();
```

Comment:
https://github.com/autowarefoundation/autoware.universe/blob/e179ad1911f497837075ec104a106f3a1b525e48/common/signal_processing/src/lowpass_filter_1d.cpp

LowpassFilter1d is very easy to use.
Assuming that new data is `u`, we just do
```
x = lpf.filter(u);
```
, then filtered value `x` can be got.

Is it possible to use butterworth filter like this? Then butterworth filter can be easily applied to other planning/control packages.

Reply:
Yes, I am implementing it in sysid package, filter, and forward backward filter methods. This package just aimed to compute the [b, a] coefficients when it was requested from me.  After I complete the filter and filtfilt methods in sysid, I will move the methods to this package.

Reply:
Okay, you will make another PR for it?